### PR TITLE
ath10k-ct: use 5.15 version

### PR DESCRIPTION
**Pull request by Ansuel**
We switched to mac80211 5.15 backport version.
Also switch ath10k-ct to 5.15 and drop the mac address patch
that got merged upstream.
Compile and tested on ipq806x Netgear R7800.
Also update the ath10k-ct to latest version to fix a typo
for the new version in the kernel log.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

File: package/kernel/ath10k-ct/Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ PKG_MIRROR_HASH:=62a1b97089d3561730656ef73beb3cd77231ec636645115cc1bbb3c6c84a6fe
 # Build the 5.10 ath10k-ct driver version.
```
Comment (robho) [inline package/kernel/ath10k-ct/Makefile:15]: Maybe remove the version from this line..

**Comment by hauke** [comment]
Could people who tested this on boards please report back if it works fine for them.

**Comment by lister-wrt** [comment]
Tested this on unifi-ac-pro for a day or so. No issues.

Something odd I noticed, it's reporting ath10k 5.12 driver. Not sure if this is just superficial or indication of an issue,

> [   22.560232] Loading modules backported from Linux version v5.15-rc6-0-g519d81956ee2
[   22.568212] Backport generated by backports.git v5.15-rc6-1-0-gd44432d6
[   22.611542] random: crng init done
[   22.649365] xt_time: kernel timezone is -0000
**[   23.023638] ath10k 5.12 driver, optimized for CT firmware, probing pci device: 0x3c.**
[   23.048850] ath10k_pci 0000:00:00.0: enabling device (0000 -> 0002)
[   23.057580] ath10k_pci 0000:00:00.0: pci irq legacy oper_irq_mode 1 irq_mode 0 reset_mode 0
[   25.310577] ath10k_pci 0000:00:00.0: qca988x hw2.0 target 0x4100016c chip_id 0x043202ff sub 0000:0000
[   25.320174] ath10k_pci 0000:00:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   25.332381] ath10k_pci 0000:00:00.0: firmware ver 10.1-ct-8x-__fH-022-ecad3248 api 2 features wmi-10.x,mfp,txstatus-noack,wmi-10.x-CT,ratemask-CT,txrate-CT,get-temp-CT,tx-rc-CT,cust-stats-CT,retry-gt2-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT crc32 1b2a161c
[   25.769580] ath10k_pci 0000:00:00.0: board_file api 1 bmi_id N/A crc32 bebc7c08
[   26.753243] ath10k_pci 0000:00:00.0: 10.1 wmi init: vdevs: 16  peers: 127  tid: 256
[   26.770721] ath10k_pci 0000:00:00.0: wmi print 'P 128 V 8 T 410'
[   26.777156] ath10k_pci 0000:00:00.0: wmi print 'msdu-desc: 1424  sw-crypt: 0 ct-sta: 0'
[   26.785457] ath10k_pci 0000:00:00.0: wmi print 'alloc rem: 25000 iram: 38944'
[   26.853683] ath10k_pci 0000:00:00.0: htt-ver 2.2 wmi-op 2 htt-op 2 cal file max-sta 128 raw 0 hwcrypto 1
[   26.863913] ath10k_pci 0000:00:00.0: NOTE:  Firmware DBGLOG output disabled in debug_mask: 0x10000000
[   27.001337] ath: EEPROM regdomain sanitized
[   27.001351] ath: EEPROM regdomain: 0x64
[   27.001355] ath: EEPROM indicates we should expect a direct regpair map
[   27.001375] ath: Country alpha2 being used: 00
[   27.001379] ath: Regpair used: 0x64
[   27.069662] ath: EEPROM regdomain sanitized
[   27.069678] ath: EEPROM regdomain: 0x64
[   27.069682] ath: EEPROM indicates we should expect a direct regpair map
[   27.069703] ath: Country alpha2 being used: 00
[   27.069707] ath: Regpair used: 0x64

**Comment by K900** [comment]
Tested on Archer C6 v2, perf seems slightly better than the 5.10 branch, but within test variance.

**Comment by castiel652** [comment]
> Something odd I noticed, it's reporting as ath10k 5.12 driver. Not sure if this is just superficial or indication of an issue,
> 
> > **[   23.023638] ath10k 5.12 driver, optimized for CT firmware, probing pci device: 0x3c.**
Ben probably forgot to update it. Not really a big issue.

**Comment by luaraneda** [comment]
> Ben probably forgot to update it. Not really a big issue.

Exactly.
Line at current HEAD on master branch (note path is for 5.15 kernel):
https://github.com/greearb/ath10k-ct/blob/3d3450c55b983c22753a0f7ff197ec1525140348/ath10k-5.15/pci.c#L3877
`printk(KERN_INFO "ath10k 5.12 driver, optimized for CT firmware, probing pci device: 0x%x.\n",`

**Comment by Ansuel** [comment]
Opened issue and he just fixed
https://github.com/greearb/ath10k-ct/commit/e8e166fee23226e5d495afbdb7ed681d747d115b

**Comment by Ansuel** [comment]
@hauke i also updated this to address the typo in the kernel message.

File: package/kernel/ath10k-ct/Makefile [hunk 2]
```diff
@@ -8,14 +8,14 @@ PKG_LICENSE_FILES:=
 
 PKG_SOURCE_URL:=https://github.com/greearb/ath10k-ct.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2021-09-22
-PKG_SOURCE_VERSION:=e6a7d5b5b834737cd12e357b5efdc2e42d923bf6
-PKG_MIRROR_HASH:=62a1b97089d3561730656ef73beb3cd77231ec636645115cc1bbb3c6c84a6fe3
+PKG_SOURCE_DATE:=2021-11-21
+PKG_SOURCE_VERSION:=e8e166fee23226e5d495afbdb7ed681d747d115b
+PKG_MIRROR_HASH:=f455df42e42c49381cbc3b8e5b510d877e2f9974641cccb73408f0c5b96c34ce
```
Comment (hauke) [inline package/kernel/ath10k-ct/Makefile:13]: This hash looks wrong to me, I am getting this:
```
Hash of the local file ath10k-ct-2021-11-21-e8e166fe.tar.xz does not match (file: 771193292242fd8c198e9e379aa90882b1f5a144fbd77e09fbc97aca8ffe84e1, requested: f455df42e42c49381cbc3b8e5b510d877e2f9974641cccb73408f0c5b96c34ce) 
```

Comment (Ansuel) [reply package/kernel/ath10k-ct/Makefile:13]: will update but strange i extracted the hash from the same warning when i updated the package...

**Comment by anomeome** [comment]
WFM on a C7V2

**Comment by hauke** [comment]
The hash also contains the PKG_SOURCE_DATE variable, I sometimes update the `PKG_SOURCE_DATE` variable after setting the hash and then it is wrong.

**Comment by hauke** [comment]
Thank you for the patch, I applied it to master in 553a3ac221e92c6fad9ffa9229b18a4e77e42c23.

**Pull request closed without merging**